### PR TITLE
Make System.Formats.Asn1 triple slash documentation the source of truth

### PR DESCRIPTION
**Pull request by vcsjones**
The pull request makes the triple slash documentation for System.Formats.Asn1 the source of truth for documentation.

1. Use the `PortToTripleSlash` tool to update the C# documentation with the latest documentation from dotnet-api-docs.

    This made substantial changes that were largely formatting, which I reverted. I kept the changes to two things: actual content changes (text changes, missing documentation, etc) and re-ordering XML elements as appropriate (param elements not matching parameter order, putting remarks at the end)

2. Remove `<UseCompilerGeneratedDocXmlFile>false</UseCompilerGeneratedDocXmlFile>`. The default value for this is true. I can't remember where, but I recall that our preference is to just remove it entirely, not explicitly set it to true.

### CODE REVIEW

**Comment by vcsjones** [comment]
cc @carlossanlop in case I missed anything.

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs [hunk 1]
```diff
@@ -41,15 +41,15 @@ public static partial class AsnDecoder
         ///   <paramref name="ruleSet"/> is not defined.
         /// </exception>
         /// <exception cref="AsnContentException">
-        ///   the next value does not have the correct tag.
+        ///   The next value does not have the correct tag.
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs:44]: These might need `<para>` tags so they remain on separate lines in the docs, right @carlossanlop?

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs [hunk 2]
```diff
@@ -151,6 +146,11 @@ public static bool TryReadPrimitiveBitString(
         ///
         ///   <paramref name="destination"/> overlaps <paramref name="source"/>.
         /// </exception>
+        /// <remarks>
+        ///   The least significant bits in the last byte which are reported as "unused" by the
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs:150]:
```suggestion
        ///   The least significant bits in the last byte that are reported as "unused" by the
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs [hunk 3]
```diff
@@ -244,31 +244,31 @@ public static bool TryReadBitString(
...
+        ///   The contents are not valid under the current encoding rules.
         /// </exception>
         /// <exception cref="ArgumentException">
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagClass"/> is
         ///   <see cref="TagClass.Universal"/>, but
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        /// <remarks>
+        ///   The least significant bits in the last byte which are reported as "unused" by the
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs:268]:
```suggestion
        ///   The least significant bits in the last byte that are reported as "unused" by the
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Enumerated.cs [hunk 4]
```diff
@@ -307,31 +307,27 @@ public ReadOnlyMemory<byte> ReadEnumeratedBytes(Asn1Tag? expectedTag = null)
         ///   Reads the next value as an Enumerated with a specified tag, converting it to the
         ///   non-[<see cref="FlagsAttribute"/>] enum specified by <typeparamref name="TEnum"/>.
         /// </summary>
+        /// <typeparam name="TEnum">Destination enum type.</typeparam>
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Enumerated.cs:310]:
```suggestion
        /// <typeparam name="TEnum">The destination enum type.</typeparam>
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.NamedBitList.cs [hunk 5]
```diff
@@ -405,25 +405,25 @@ public partial class AsnReader
         ///   Reads the next value as a NamedBitList with a specified tag, converting it to the
         ///   [<see cref="FlagsAttribute"/>] enum specified by <typeparamref name="TFlagsEnum"/>.
         /// </summary>
+        /// <typeparam name="TFlagsEnum">Destination enum type.</typeparam>
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.NamedBitList.cs:408]:
```suggestion
        /// <typeparam name="TFlagsEnum">The destination enum type.</typeparam>
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Sequence.cs [hunk 6]
```diff
@@ -27,31 +27,31 @@ public static partial class AsnDecoder
...
         /// <exception cref="ArgumentException">
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagClass"/> is
         ///   <see cref="TagClass.Universal"/>, but
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        /// <remarks>
+        ///   The nested content is not evaluated by this method, except for minimal processing to
+        ///   determine the location of an end-of-contents marker.
+        ///   Therefore, the contents may contain data which is not valid under the current encoding rules.
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Sequence.cs:53]:
```suggestion
        ///   Therefore, the contents might contain data that's not valid under the current encoding rules.
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Sequence.cs [hunk 7]
```diff
@@ -109,27 +109,27 @@ public partial class AsnReader
...
         /// </exception>
         /// <exception cref="ArgumentException">
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagClass"/> is
         ///   <see cref="TagClass.Universal"/>, but
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        /// <remarks>
+        ///   The nested content is not evaluated by this method, and may contain data
+        ///   which is not valid under the current encoding rules.
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Sequence.cs:131]:
```suggestion
        ///   The nested content is not evaluated by this method and might contain data
        ///   that's not valid under the current encoding rules.
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.SetOf.cs [hunk 8]
```diff
@@ -32,32 +32,32 @@ public static partial class AsnDecoder
...
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagClass"/> is
         ///   <see cref="TagClass.Universal"/>, but
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        /// <remarks>
+        ///   The nested content is not evaluated by this method, except for minimal processing to
+        ///   determine the location of an end-of-contents marker or verification of the content
+        ///   sort order.
+        ///   Therefore, the contents may contain data which is not valid under the current encoding rules.
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.SetOf.cs:59]:
```suggestion
        ///   Therefore, the contents might contain data that's not valid under the current encoding rules.
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.SetOf.cs [hunk 9]
```diff
@@ -143,27 +143,27 @@ public partial class AsnReader
...
         /// </exception>
         /// <exception cref="ArgumentException">
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagClass"/> is
         ///   <see cref="TagClass.Universal"/>, but
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        /// <remarks>
+        ///   The nested content is not evaluated by this method (aside from sort order, when
+        ///   required), and may contain data which is not valid under the current encoding rules.
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.SetOf.cs:165]:
```suggestion
        ///   required) and might contain data that's not valid under the current encoding rules.
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.SetOf.cs [hunk 10]
```diff
@@ -186,27 +186,27 @@ public AsnReader ReadSetOf(Asn1Tag? expectedTag = null)
...
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.SetOf.cs:208]:
```suggestion
        ///   required) and might contain data that's not valid under the current encoding rules.
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.OctetString.cs [hunk 11]
```diff
@@ -53,6 +47,12 @@ public Scope PushOctetString(Asn1Tag? tag = null)
         /// <exception cref="InvalidOperationException">
         ///   the writer is not currently positioned within an Octet String with the specified tag.
         /// </exception>
+        /// <remarks>
+        ///   In <see cref="AsnEncodingRules.BER"/> and <see cref="AsnEncodingRules.DER"/> modes
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.OctetString.cs:51]:
```suggestion
        ///   In <see cref="AsnEncodingRules.BER"/> and <see cref="AsnEncodingRules.DER"/> modes,
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.OctetString.cs [hunk 12]
```diff
@@ -53,6 +47,12 @@ public Scope PushOctetString(Asn1Tag? tag = null)
...
+        ///   the encoded contents will remain in a single primitive Octet String.
+        ///   In <see cref="AsnEncodingRules.CER"/> mode the contents will be broken up into
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.OctetString.cs:53]:
```suggestion
        ///   In <see cref="AsnEncodingRules.CER"/> mode, the contents will be broken up into
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.SetOf.cs [hunk 13]
```diff
@@ -14,16 +14,16 @@ public sealed partial class AsnWriter
...
-        ///   the writer will sort the Set-Of elements when the tag is closed.
-        /// </remarks>
         /// <exception cref="ArgumentException">
         ///   <paramref name="tag"/>.<see cref="Asn1Tag.TagClass"/> is
         ///   <see cref="TagClass.Universal"/>, but
         ///   <paramref name="tag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        /// <remarks>
+        ///   In <see cref="AsnEncodingRules.CER"/> and <see cref="AsnEncodingRules.DER"/> modes
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.SetOf.cs:24]:
```suggestion
        ///   In <see cref="AsnEncodingRules.CER"/> and <see cref="AsnEncodingRules.DER"/> modes,
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.SetOf.cs [hunk 14]
```diff
@@ -51,6 +47,10 @@ public Scope PushSetOf(Asn1Tag? tag = null)
         /// <exception cref="InvalidOperationException">
         ///   the writer is not currently positioned within a Set-Of with the specified tag.
         /// </exception>
+        /// <remarks>
+        ///   In <see cref="AsnEncodingRules.CER"/> and <see cref="AsnEncodingRules.DER"/> modes
```
Comment (gewarren) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.SetOf.cs:51]:
```suggestion
        ///   In <see cref="AsnEncodingRules.CER"/> and <see cref="AsnEncodingRules.DER"/> modes,
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs (hunk 1 above)
Comment (gewarren) [reply src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs:44]: I see now that that would mean a lot of changes throughout this PR. I'm also fine with not adding them.

**Review by gewarren — APPROVED** [review]
Thanks for doing this!

**Review by carlossanlop — APPROVED** [review]
LGTM (modulo @gewarren's suggestions). Can you please backport it to release/9.0 when you're done?

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/Asn1Tag.Accelerators.cs [hunk 15]
```diff
@@ -43,7 +43,7 @@ public partial struct Asn1Tag
             new Asn1Tag(ConstructedMask, (int)UniversalTagNumber.OctetString);
 
         /// <summary>
-        ///   Represents the universal class tag for a Null value.
+        ///   Represents the universal class tag for a <see langword="null" /> value.
```
Comment (bartonjs) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/Asn1Tag.Accelerators.cs:46]: I'm not sure this is a correct change.  It's referencing the ASN-NULL, not the .NET-null

Comment (carlossanlop) [reply src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/Asn1Tag.Accelerators.cs:46]:
```suggestion
        ///   Represents the universal class tag for a <c>Null</c> value.
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.NamedBitList.cs [hunk 16]
```diff
@@ -97,7 +97,7 @@ public static partial class AsnDecoder
         ///     }
         ///   </code>
         ///
-        ///   Note that while the example here uses the KeyUsage NamedBitList from
+        ///   While the example here uses the KeyUsage NamedBitList from
         ///   <a href="https://tools.ietf.org/html/rfc3280#section-4.2.1.3">RFC 3280 (4.2.1.3)</a>,
         ///   the example enum uses values thar are different from
```
Comment (bartonjs) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.NamedBitList.cs:102]:
```suggestion
        ///   the example enum uses values that are different from
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.NamedBitList.cs [hunk 17]
```diff
@@ -476,7 +476,7 @@ public partial class AsnReader
...
```
Comment (bartonjs) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.NamedBitList.cs:481]:
```suggestion
        ///   the example enum uses values that are different from
```

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Oid.cs [hunk 18]
```diff
@@ -25,21 +25,21 @@ public static partial class AsnDecoder
         ///   The tag to check for before reading, or <see langword="null"/> for the default tag (Universal 6).
         /// </param>
         /// <returns>
-        ///   The decoded object identifier, in dotted-decimal notation.
+        ///   The decoded object identifier in a dotted-decimal notation.
```
Comment (bartonjs) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Oid.cs:28]:
```suggestion
        ///   The decoded object identifier in the dotted-decimal notation.
```

**Review by bartonjs — APPROVED** [review]

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.BitString.cs:44]: The porttotripleslash tool did not put \<para\> there, so I assumed it was not required.

**Comment by gewarren** [comment]
@carlossanlop Is there a documented set of steps needed to set this repo as the source of truth? If so, can you add a step to disallow edits in the dotnet-api-docs repo for the affected namespace? Like [this](https://github.com/dotnet/dotnet-api-docs/pull/10583).

**Comment by ViktorHofer** [comment]
>  The default value for this is true. I can't remember where, but I recall that our preference is to just remove it entirely, not explicitly set it to true.

Correct 👍 and big thanks for doing this.

**Pull request merged**